From a0105915c6dfd743fd925724a46345e7bebc6326 Mon Sep 17 00:00:00 2001
From: Finley Xiao <finley.xiao@rock-chips.com>
Date: Thu, 25 May 2017 11:51:11 +0800
Subject: [PATCH] PM / devfreq: assign set_auto_self_refresh for rk3368

This makes it possible to enter ddr self-refresh mode
when early suspend.

Change-Id: Ib72f391af00674a3c3ab32bbbd4e4a857d3354e8
Signed-off-by: Finley Xiao <finley.xiao@rock-chips.com>
---
 drivers/devfreq/rockchip_dmc.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/devfreq/rockchip_dmc.c b/drivers/devfreq/rockchip_dmc.c
index f07f06db8d8b..0a6b993bc7a5 100644
--- a/drivers/devfreq/rockchip_dmc.c
+++ b/drivers/devfreq/rockchip_dmc.c
@@ -724,7 +724,8 @@ static int rk3288_dmc_init(struct platform_device *pdev)
 	return 0;
 }
 
-static int rk3368_dmc_init(struct platform_device *pdev)
+static int rk3368_dmc_init(struct platform_device *pdev,
+			   struct rockchip_dmcfreq *dmcfreq)
 {
 	struct device *dev = &pdev->dev;
 	struct device_node *np = pdev->dev.of_node;
@@ -792,6 +793,8 @@ static int rk3368_dmc_init(struct platform_device *pdev)
 	else
 		dev_dbg(dev, ("%s out\n"), __func__);
 
+	dmcfreq->set_auto_self_refresh = scpi_ddr_set_auto_self_refresh;
+
 	return 0;
 }
 
-- 
2.35.3

